src/main/kotlin/StringAddCalculator.kt

```diff
@@ -0,0 +1,43 @@
+class StringAddCalculator {
+    fun add(text: String?): Int {
+        if (text.isNullOrEmpty()) {
+            return 0
+        }
+        if (text.contains("-")) {
+            throw RuntimeException("음수는 사용할 수 없습니다")
+        }
+        val (plainText, customSeparator) = separateCustomSeparatorWithText(text)
+        val replacedOperator = replaceToSameSeparator(plainText, customSeparator)
+        val numbers = replacedOperator.split(COMMON_SEPARATOR).map { it.toInt() }
+        return numbers.sum()
+    }
+
+    private fun replaceToSameSeparator(text: String, customSeparator: String?): String {
+        var replacedOperator = text
+        if (customSeparator != null) {
+            replacedOperator = replacedOperator.replace(customSeparator, COMMON_SEPARATOR)
+        }
+        DEFAULT_SEPARATORS.forEach {
+            replacedOperator = replacedOperator.replace(it, COMMON_SEPARATOR)
+        }
+        return replacedOperator
```

Comment:
```suggestion
        if (customSeparator != null) {
            return text.replace(customSeparator, COMMON_SEPARATOR)
        }

        return text.replace(":", COMMON_SEPARATOR)
```
early return과 로직을 활용하면 간단하게 코드를 작성할 수 있을 것 같습니다 :)